### ISSUE TITLE
Improve S5122: raise only for permissive access (*) 

### ISSUE THREAD

**Issue by eric-therond-sonarsource** [SonarSource/SonarJS#1837]
### Raise only for permissive access (*)
[S5122](https://jira.sonarsource.com/browse/RSPEC-5122) should be reworked to raise only when permissive CORS policy is obvious (`'Access-Control-Allow-Origin': '*'`)

Following code should not raise an issue
```
const srv2 = http.createServer((req, res) => {
	res.setHeader('Access-Control-Allow-Origin', 'http://mytrustedorigin'); // FP
```

### Add support for Express.js API

Code examples: https://github.com/SonarSource/security-expected-issues/tree/master/javascript/rules/hotspots/RSPEC-5122

For the `cors` middleware raise an issue when the `origin` property of the options is set to `*`:
```
let corsOptions = {
  origin: '*' // Sensitive (main location)
};

let app = express();
app.use(cors(corsOptions));  // Sensitive (secondary location on the cors() call)
```
Without options passed in the call (ie: `cors()`) the code is sensitive (equals to `origin` set to `*`)

Related issue:
https://github.com/SonarSource/SonarJS/issues/1145

**Comment by saberduck** [comment]
We should also support `res.set()` http://expressjs.com/en/api.html#res.set , which is alias to `res.header()`

**Comment by saberduck** [comment]
Secondary locations on hotspots will be supported by https://jira.sonarsource.com/browse/MMF-2084
